app/controllers/registration/registration.controller.js

```diff
@@ -135,18 +137,23 @@ function submitChooseSignInMethodPage (req, res) {
     })
   }
 
-  if (signInMethod === 'SMS') {
-    res.redirect(paths.register.phoneNumber)
-  } else {
-    res.redirect(paths.register.authenticatorApp)
+  try {
+    await adminusersClient.reprovisionOtp(sessionData.code)
+    if (signInMethod === 'SMS') {
```

Comment:
Another mechanism here to protect against some of the edge cases described in the commit message might be to keep track of what the last method that an OTP was provisioned against was (set anytime that we call this admin users endpoint) and then only re-provision if it's different, something along the lines of

```javascript
if (signInMethod != sessionData.signInMethod) {
  await adminusersClient.reprovisionOtp(sessionData.code)
}
```

Reply:
Ah yes good idea - that might be preferable. I guess we could go back to re-provisioning in the old places but use this cookie to decide.